### DISCUSSION TITLE
Add CSV customization when adding documents (like delimiter)

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#541] [Feedback & Feature Proposal]
Following https://github.com/meilisearch/meilisearch/issues/2806

Meilisearch accepts CSV dataset, but in a specific format. The users seem to need to customize the format rules of the CSV, like the delimiter

A first solution has been suggested in this PR: https://github.com/meilisearch/meilisearch/pull/2803

**Comment by MixusMinimax** [comment]
At that point it might even be better to parse the entire csv::readerbuilder from query arguments using serde for example, and this could maybe even apply to json, though there isn't much configurability to json (apart from maybe comments), but I do have to say that delimiters are very commonly something different from commas
